File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/DriverInformation.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 驾驶员信息实体类，对应数据库表 "driver_information"
 * 包含驾驶员的基本信息以及驾驶证相关信息
 */
@Data
@TableName("driver_information")
public class DriverInformation implements Serializable {
    /**
     * 序列化版本 UID，用于对象序列化
     */
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 驾驶员 ID，主键，自动生成
     */
    @TableId(value = "driver_id", type = IdType.AUTO)
    private Integer driverId;

    /**
     * 姓名
     */
    @TableField("name")
    private String name;

    /**
     * 身份证号码
     */
    @TableField("id_card_number")
    private String idCardNumber;

    /**
     * 联系电话
     */
    @TableField("contact_number")
    private String contactNumber;

    /**
     * 驾驶证号码
     */
    @TableField("driver_license_number")
    private String driverLicenseNumber;

    /**
     * 性别
     */
    @TableField("gender")
    private String gender;

    /**
     * 出生日期
     */
    @TableField("birthdate")
    private LocalDate birthdate;

    /**
     * 首次领取驾驶证日期
     */
    @TableField("first_license_date")
    private LocalDate firstLicenseDate;

    /**
     * 允许驾驶的车辆类型
     */
    @TableField("allowed_vehicle_type")
    private String allowedVehicleType;

    /**
     * 驾驶证发证日期
     */
    @TableField("issue_date")
    private LocalDate issueDate;

    /**
     * 驾驶证有效期截止日期
     */
    @TableField("expiry_date")
    private LocalDate expiryDate;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}

File: final_asssignment_backend_go/project/internal/domain/driver_information.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// DriverInformation 表示 driver_information 表的实体
type DriverInformation struct {
	DriverID            int            `gorm:"column:driver_id;primaryKey;autoIncrement" json:"driver_id"`
	Name                string         `gorm:"column:name" json:"name"`
	IDCardNumber        string         `gorm:"column:id_card_number" json:"id_card_number"`
	ContactNumber       string         `gorm:"column:contact_number" json:"contact_number"`
	DriverLicenseNumber string         `gorm:"column:driver_license_number" json:"driver_license_number"`
	Gender              string         `gorm:"column:gender" json:"gender"`
	Birthdate           time.Time      `gorm:"column:birthdate" json:"birthdate"`
	FirstLicenseDate    time.Time      `gorm:"column:first_license_date" json:"first_license_date"`
	AllowedVehicleType  string         `gorm:"column:allowed_vehicle_type" json:"allowed_vehicle_type"`
	IssueDate           time.Time      `gorm:"column:issue_date" json:"issue_date"`
	ExpiryDate          time.Time      `gorm:"column:expiry_date" json:"expiry_date"`
	DeletedAt           gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (DriverInformation) TableName() string {
	return "driver_information"
}
